### PR TITLE
mafia install <package>

### PR DESCRIPTION
**Pull request by jacobstanley**
This allows the use of mafia's global cache for installing executables from Hackage.

Usage:

```
$ mafia install pretty-show
stderr> Building...
stderr> Building...
stdout> /Users/jdoe/.ambiata/mafia/bin/pretty-show/bin
```

Or even better:

```
$ echo "Foo(Bar     (Baz 1 2    3)   )" | $(mafia install pretty-show)/ppsh
Foo (Bar (Baz 1 2 3))
```

It also works with a version number suffix:

```
$ echo "[Max\n1,ML\n2]" | $(mafia install pretty-show-1.6.9)/ppsh
[ Max 1, ML 2 ]
```

The existing `MAFIA_HAPPY`, `MAFIA_ALEX` things now use the global cache as a result of this change.

### CODE REVIEW

**Comment by tmcgilchrist** [comment]
:+1: looks really useful.

File: src/Mafia/Bin.hs [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+    env <- getPackageEnv
+    let gdir = packageSandboxDir env pkg
+    liftIO $ createSymbolicLink (T.unpack gdir) (T.unpack plink)
+
+  return pbin
+
+ensureExeOnPath :: InstallPackage -> EitherT InstallError IO ()
+ensureExeOnPath pkg = do
+  dir <- installBinary pkg
+  setEnv "PATH" . maybe dir (\path -> dir <> ":" <> path) =<< lookupEnv "PATH"
```
Comment (charleso) [inline src/Mafia/Bin.hs:109]: I know I added this, but I wish I'd done something a little safer - like creating a `Exe` data type which you pass around.

**Comment by charleso** [comment]
:+1:

File: src/Mafia/Bin.hs (hunk 1 above)
Comment (jacobstanley) [inline src/Mafia/Bin.hs:109]: so you just put it on the path temporarily?

**Comment by jacobstanley** [comment]
I noticed you'd have a strange situation if you install `async` for example. I added a small patch to show an error that the bin/ directory cannot be linked as there are no executables. The package will still be installed however.

**Pull request merged**